python/test/t_QuadraticLeastSquares_std.py

```diff
@@ -1,54 +1,63 @@
 #! /usr/bin/env python
 
 import openturns as ot
+import openturns.testing as ott
 
-
-myFunc = ot.SymbolicFunction(
+f = ot.SymbolicFunction(
     ["x1", "x2"], ["x1*sin(x2)", "cos(x1+x2)", "(x2+1)*exp(x1-2*x2)"]
 )
-data = ot.Sample(9, myFunc.getInputDimension())
-point = ot.Point(myFunc.getInputDimension())
-point[0] = 0.5
-point[1] = 0.5
-data[0] = point
-point[0] = -1
-point[1] = -1
-data[1] = point
-point[0] = -1
-point[1] = 1
-data[2] = point
-point[0] = 1
-point[1] = -1
-data[3] = point
-point[0] = 1
-point[1] = 1
-data[4] = point
-point[0] = -0.5
-point[1] = -0.5
-data[5] = point
-point[0] = -0.5
-point[1] = 0.5
-data[6] = point
-point[0] = 0.5
-point[1] = -0.5
-data[7] = point
-point[0] = 0.5
-point[1] = 0.5
-data[8] = point
-myLeastSquares = ot.QuadraticLeastSquares(data, myFunc)
-myLeastSquares.run()
-responseSurface = myLeastSquares.getMetaModel()
-print("myLeastSquares=", repr(myLeastSquares))
-print("responseSurface=", repr(responseSurface))
-inPoint = ot.Point(myFunc.getInputDimension(), 0.2)
-print("myFunc(", repr(inPoint), ")=", repr(myFunc(inPoint)))
-print("responseSurface(", repr(inPoint), ")=", repr(responseSurface(inPoint)))
-dataOut = myFunc(data)
-myLeastSquares = ot.QuadraticLeastSquares(data, dataOut)
-myLeastSquares.run()
-responseSurface = myLeastSquares.getMetaModel()
-print("myLeastSquares=", repr(myLeastSquares))
-print("responseSurface=", repr(responseSurface))
-inPoint = ot.Point(myFunc.getInputDimension(), 0.2)
-print("myFunc(", repr(inPoint), ")=", repr(myFunc(inPoint)))
-print("responseSurface(", repr(inPoint), ")=", repr(responseSurface(inPoint)))
+X = ot.Sample([[0.5] * 2, [-1.0] * 2, [-1.0, 1.0], [1.0, -1.0], [1.0] * 2, [-0.5] * 2, [-0.5, 0.5], [0.5, -0.5], [0.5] * 2])
```

Comment:
- An uppercase variable name should be avoided according to the [PEP8 for variable names](https://peps.python.org/pep-0008/#function-and-variable-names). I suggest to rename `X` into `inputSample` and `Y` into `outputSample`.
- This particular DoE seems to be a poor choice to me. We should use [Box](http://openturns.github.io/openturns/latest/user_manual/_generated/openturns.Box.html) with a number of points which exceeeds the number of parameters. With 2 inputs, the quadratic model has 1 constant parameter, 2 parameters for the linear trend and 3 parameters for the quadratic part. This is a total of 6 coefficients to estimate.
```
levels = [3, 3]  # Because 3 x 3 = 9
bounds = ot.Interval([-1.0, -1.0], [1.0, 1.0])
myGrid = ot.Box(levels, bounds)
inputSample = myGrid.generate()
```

Reply:
yes, but this gives null linear coefficients